### ISSUE TITLE
TableWorkspace display very slow when workspace has large numbers of rows

### ISSUE THREAD

**Issue by DanielMurphy22** [mantidproject/mantid#29636]
Raised in smoke testing by @DannyHindson : https://github.com/mantidproject/mantid/issues/29607#issuecomment-699878909

### Expected behavior
Show detectors for all instruments works quickly.

### Actual behavior
Showing detectors on TOPAZ_3132_event.nxs hangs

### Steps to reproduce the behavior
- Load TOPAZ_3132_event.nxs and right-click on the ws and select "Show Detectors".
- Alternatively, execute the `CreateDetectorTable` algorithm and then show data on the resulting table ws

### Platforms affected

Found on Windows, Smoke testing for release 5.1
Not a problem on 5.0

The task here is to look at improving the speed of generating the display of a TableWorkspace in Workbench that has a large number of rows. Large is of course a bit subjective but the above example seems to provide a good test case. Even if we reach a point where it takes a little time to show then we should look into ensuring that Workbench does not hang and appear unresponsive.

**Comment by DanielMurphy22** [comment]
A very similar thing was found by @Pasarus on Ubuntu (also on Windows) for LARMOR3368 on release candidate for 6.0 https://github.com/mantidproject/mantid/issues/30655#issuecomment-776057667

**Comment by martyngigg** [comment]
@peterfpeterson also mentioned on Slack that table workspaces lots of rows generally are slow. I'm going to update the description of the task to reflect that fact that the idea is to improve the speed of that here.

**Comment by peterfpeterson** [comment]
The fix with `PeaksWorkspace` in mantidplot/qt4 was relatively straightforward. What is happening is the default qt-widget for rendering the table converts *everything* that may ever be rendered to a string to determine the default width for the columns; then displays the widget with only the data currently in view converted to strings. The fix for `PeaksWorkspace` was to override the method that determines the default column width to use a fixed value without inspecting the actual data. For a "generic" view, we could set a fixed value for `int`, and look at the dynamic range of `float` to return a value. This would be significantly faster than the default qt behavior.

**Comment by peterfpeterson** [comment]
Finding some of the old implementation. The magic bit was [`QPeaksTableModel::numCharacters`](https://github.com/mantidproject/mantid/blob/v6.0.0/qt/widgets/sliceviewer/src/QPeaksTableModel.cpp#L103) which was used in [`PeaksWorkspaceWidget::createTableMVC`](https://github.com/mantidproject/mantid/blob/v6.0.0/qt/widgets/sliceviewer/src/PeaksWorkspaceWidget.cpp#L128-L156).

**Comment by DannyHindson** [comment]
I just looked back in my notes about this issue and I think the performance used to be OK in workbench (in 5.0). And something happened in 5.1 to slow it down. At the time I started to looking into whether it was something to do with changes to qt/python/mantidqt/widgets/workspacedisplay/table/presenter.py. I was looking at whether it was this commit 0ecfe1acf089b04325d3133bc5f5d49dfc4f1572 but then ran out of time to progress it further
